editor/io_plugins/editor_scene_import_plugin.cpp

```diff
@@ -1567,7 +1567,9 @@ Node *EditorSceneImportPlugin::_fix_node(Node *p_node, Node *p_root, Map<Ref<Mes
 		mi->set_baked_light_texture_id(layer);
 	}
 
-	if (p_flags & SCENE_FLAG_CREATE_COLLISIONS && _teststr(name, "colonly")) {
+	bool is_rigid = false;
+
+	if (p_flags & SCENE_FLAG_CREATE_COLLISIONS && (_teststr(name, "colonly") || (is_rigid = _teststr(name, "rigidonly") && !p_node->cast_to<MeshInstance>()))) {
```

Comment:
Would doing `bool is_rigid = _teststr(name, "rigidonly");`, then testing against `is_rigid` be better?